functions/Set-DbaAgentJobStep.ps1

```diff
@@ -0,0 +1,332 @@
+                $JobStep = $Server.JobServer.Jobs[$JobName].JobSteps[$StepName]
+
+                #region job step options
+                # Setting the options for the job step
+                if($NewName.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step name to '$($NewName)'" -Level 5 -Silent $Silent
+                    $JobStep.Rename($NewName)
+                }
+
+                if($Subsystem.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step subsystem to '$($Subsystem)'" -Level 5 -Silent $Silent
+                    $JobStep.Subsystem = $Subsystem
+                }
+
+                if($Command.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step command to '$($Command)'" -Level 5 -Silent $Silent
+                    $JobStep.Command = $Command
+                }
+
+                if($CmdExecSuccessCode -ge 0)
+                {
+                    Write-Message -Message "Setting job step command exec success code to $($CmdExecSuccessCode)" -Level 5 -Silent $Silent
+                    $JobStep.CommandExecutionSuccessCode = $CmdExecSuccessCode
+                }
+
+                if($OnSuccessAction.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step success action to '$($OnSuccessAction)'" -Level 5 -Silent $Silent
+                    $JobStep.OnSuccessAction = $OnSuccessAction
+                }
+
+                if($OnSuccessStepId -ge 1)
+                {
+                    Write-Message -Message "Setting job step success step id to $($OnSuccessStepId)" -Level 5 -Silent $Silent
+                    $JobStep.OnSuccessStep = $OnSuccessStepId
+                }
+
+                if($OnFailAction.Length -ge 1)
+                {
+                    Write-Message -Message "Setting job step fail action to '$($OnFailAction)'" -Level 5 -Silent $Silent
+                    $JobStep.OnFailAction = $OnFailAction
+                }
+
+                if($OnFailStepId -ge 1)
+                {
+                    Write-Message -Message "Setting job step fail step id to $($OnFailStepId)" -Level 5 -Silent $Silent
+                    $JobStep.OnFailStep = $OnFailStepId
+                }
+
+                if($DatabaseName.Length -ge 1)
+                {
+                    # Check if the database is present on the server
+                    if(($Server.Databases).Name -contains $DatabaseName)
+                    {
+                        Write-Message -Message "Setting job step database name to '$($DatabaseName)'" -Level 5 -Silent $Silent
+                        $JobStep.DatabaseName = $DatabaseName
+                    }
+                    else 
+                    {
+                        Write-Message -Message ("The database is not present on instance '$($SqlServer)'.") -Warning -Silent $Silent 
+                        return
```

Comment:
Unneeded braces around the string.
The `-Warning` parameter has been deprecated by Write-Message (it will write a warning and tell you so if you run it). Use `-Level Warning` instead.
Also if you are interrupting an execution for this, consider using `Stop-Function` instead. Is it worth interrupting the flow for this?